forest/src/cli/chain_cmd.rs

```diff
@@ -48,6 +69,82 @@ impl ChainCommands {
                 let cid: Cid = cid.parse().unwrap();
                 print_rpc_res_pretty(chain_get_block((CidJson(cid),)).await);
             }
+            Self::Export {
+                tipset,
+                recent_stateroots,
+                skip_old_messages,
+                output_path,
+            } => {
+                let chain_finality = Policy::default().chain_finality;
+                let recent_stateroots = match recent_stateroots {
+                    Some(rsrs) => {
+                        let rsrs = rsrs.to_owned();
+                        if rsrs < chain_finality {
+                            return cli_error_and_die(
+                                &format!(
+                                    "\recent-stateroots\" must be greater than {}",
+                                    chain_finality
+                                ),
+                                1,
+                            );
+                        }
+
+                        if rsrs == 0 && *skip_old_messages {
+                            return cli_error_and_die(
+                                "must pass recent stateroots along with skip-old-messages",
+                                1,
+                            );
+                        }
+
+                        rsrs
+                    }
+                    None => 0,
+                };
+
+                let chain_head = match chain_head().await {
+                    Ok(head) => head.0,
+                    Err(_) => return cli_error_and_die("Could not get network head", 1),
+                };
+
+                let output_path = match output_path {
+                    Some(path) => path.to_owned(),
+                    None => {
+                        let now = OffsetDateTime::now_utc();
+                        format!(
+                            "forest_snapshot_{}_{}_{}_height_{}.car",
+                            now.year(),
+                            now.month(),
+                            now.day(),
+                            chain_head.epoch()
+                        )
+                    }
+                };
+
+                let epoch = if let Some(epoch) = tipset {
+                    *epoch
+                } else {
+                    chain_head.epoch()
+                };
+
+                if recent_stateroots == 0 && *skip_old_messages {
+                    return cli_error_and_die(
+                        "Must pass recent stateroots along with skip-old-messages",
+                        1,
+                    );
+                }
+
+                let params = (
+                    epoch,
+                    recent_stateroots,
+                    *skip_old_messages,
+                    output_path.clone(),
+                    TipsetKeysJson(chain_head.key().clone()),
+                );
+
+                chain_export(params).await.map_err(handle_rpc_err).unwrap();
```

Comment:
`unwrap` is mean

Reply:
do we get info where the file was saved?

Reply:
No, however I can make the API return the output file as a pathbuf to be displayed to the user